### PR TITLE
Make OpenAIRE Funding connector mapping configurable

### PR DESCRIPTION
**Pull request by LucaGiamminonni**
## Description
This PR improves the implementation of org.dspace.external.provider.impl.OpenAIREFundingDataProvider to make metadata field mapping configurable.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [X] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [X] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [X] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [X] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [X] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [X] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Review by paulo-graca — APPROVED** [review]
Thanks @LucaGiamminonni  for adding these code optimizations and flexibility. I have no code suggestions. And since this PR is passing existing tests, that's a +1 on my side.

File: dspace/config/spring/api/external-openaire.xml [hunk 1]
```diff
@@ -10,17 +14,75 @@
...
+    <util:map id="mapOfmetadata"
+        map-class="java.util.HashMap" key-type="java.lang.String" value-type="org.dspace.importer.external.metadatamapping.MetadataFieldConfig">
+        <entry key="awardNumber" value-ref="openaireAwardNumber" />
+        <entry key="fundingStream" value-ref="openaireFundingStream" />
+        <entry key="awardURI" value-ref="openaireAwardURI" />
+        <entry key="funderName" value-ref="openaireFunderName" />
+        <entry key="funderIdentifier" value-ref="openaireFunderIdentifier" />
+        <entry key="dcTitle" value-ref="openaireTitle" />
+        <entry key="titleAlternative" value-ref="openaireTitleAlternative" />
+        <entry key="dcIdentifier" value-ref="openaireIdentifier" />
```
Comment (tdonohue) [inline dspace/config/spring/api/external-openaire.xml:38]: `dcIdentifier` doesn't appear to be used or defined in `OpenAIREFundingDataProvider`.  So, I don't think this setting does anything right now.

File: dspace/config/spring/api/external-openaire.xml [hunk 2]
```diff
@@ -10,17 +14,75 @@
...
+        <entry key="coverageSpatial" value-ref="openaireSpatial" />
+        <entry key="dcDescription" value-ref="openaireDescription" />
```
Comment (tdonohue) [inline dspace/config/spring/api/external-openaire.xml:40]: Same here, `dcDescription` doesn't appear to be used or defined in `OpenAIREFundingDataProvider`.  So, I don't think this setting does anything right now.

**Review by tdonohue — COMMENTED** [review]
@LucaGiamminonni : Overall, this works for me & it looks good. However, I found two of your new configurations are not being used.  Maybe you meant to add them to `OpenAIREFundingDataProvider` (see my notes inline below)?  If so, once you add them, I think this is ready to merge.

**Comment by tdonohue** [comment]
@LucaGiamminonni : If you have a chance, could you make the minor updates to this PR requested back in July?  This PR is basically ready to go, but I pointed out in [my last review](https://github.com/DSpace/DSpace/pull/8281#pullrequestreview-1053108778) that it looks like some configs may be overlooked or unused.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @LucaGiamminonni !  This now looks good to me.  I gave it one final test today and it's working well

**Pull request merged**